src/main.rs

```diff
@@ -299,29 +304,22 @@ fn configure_app(app: &mut web::ServiceConfig, conf: &MiniserveConfig) {
         let dirs_first = conf.dirs_first;
         let hide_version_footer = conf.hide_version_footer;
         let title = conf.title.clone();
-        upload_route = if let Some(random_route) = conf.random_route.clone() {
-            format!("/{}/upload", random_route)
-        } else {
-            "/upload".to_string()
-        };
+
         if path.is_file() {
             None
-        } else if let Some(index_file) = &conf.index {
-            Some(
-                actix_files::Files::new(&full_route, path).index_file(index_file.to_string_lossy()),
-            )
         } else {
             let u_r = upload_route.clone();
-            let files;
-            if show_hidden {
-                files = actix_files::Files::new(&full_route, path)
-                    .show_files_listing()
-                    .use_hidden_files();
-            } else {
-                files = actix_files::Files::new(&full_route, path).show_files_listing();
-            }
-
+            let files = actix_files::Files::new(&full_route, path);
+            let files = match &conf.index {
```

Comment:
Would be great if you could document this whole block of logic while you're here. :)

Reply:
I have more upcoming refactoring for this, which should make it clearer.

Reply:
Alright, that works for me, too, just make sure to keep track of it. :P